jicofo-common/src/main/java/org/jitsi/protocol/xmpp/AbstractOperationSetJingle.java

```diff
@@ -302,68 +310,6 @@ private List<ContentPacketExtension> encodeSources(
         return ret;
     }
 
-    /**
-     * The logic for processing received <tt>JingleIQ</tt>s.
-     *
-     * @param iq the <tt>JingleIQ</tt> to process.
-     */
-    private IQ processJingleIQ(JingleIQ iq, @NotNull JingleSession session)
-    {
-        JingleAction action = iq.getAction();
-
-        if (action == null)
-        {
-            // bad-request
-            return IQ.createErrorResponse(iq, StanzaError.getBuilder(StanzaError.Condition.bad_request).build());
-        }
-        stats.stanzaReceived(action);
-
-        JingleRequestHandler requestHandler = session.getRequestHandler();
-        StanzaError error = null;
-        switch (action)
-        {
-        case SESSION_ACCEPT:
-            error = requestHandler.onSessionAccept(session, iq.getContentList());
-            break;
-        case SESSION_INFO:
-            error = requestHandler.onSessionInfo(session, iq);
-            break;
-        case SESSION_TERMINATE:
-            error = requestHandler.onSessionTerminate(session, iq);
-            break;
-        case TRANSPORT_ACCEPT:
-            error = requestHandler.onTransportAccept(session, iq.getContentList());
-            break;
-        case TRANSPORT_INFO:
-            requestHandler.onTransportInfo(session, iq.getContentList());
-            break;
-        case TRANSPORT_REJECT:
-            requestHandler.onTransportReject(session, iq);
-            break;
-        case ADDSOURCE:
-        case SOURCEADD:
-            error = requestHandler.onAddSource(session, iq.getContentList());
-            break;
-        case REMOVESOURCE:
-        case SOURCEREMOVE:
-            error = requestHandler.onRemoveSource(session, iq.getContentList());
-            break;
-        default:
-            error = StanzaError.getBuilder(StanzaError.Condition.feature_not_implemented).build();
-            logger.warn("unsupported action " + action);
-        }
-
-        // FIXME IQ type is not taken into account
```

Comment:
Is this fixme now fixed with the logic rearrangement?

Reply:
I don't understand what this comment refers to. The instance type is checked to be a JingleIQ. The IQ type is always "set" because that's the type of the AbstractIqRequestHandler instance.